### ISSUE TITLE
CentOS 8 base image reaches EOL on December 31st, 2021

### ISSUE THREAD

**Issue by kstephan-wescale** [Kong/docker-kong#530]
`centos:8` is currently used as base image and [reaches EOL at the end of December](https://www.centos.org/centos-linux-eol/). CentOS 7 may be a simple alternative, it’s supported until 2024.

**Comment by agaffney** [comment]
FYI, the Kong image based on CentOS can no longer perform yum operations as-is. The latest upstream `centos:8` image suffers from the same problem.

```
$ yum ...
...
CentOS Linux 8 - AppStream                      136  B/s |  38  B     00:00    
Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
```

**Comment by curiositycasualty** [comment]
A near-term workaround was just merged in here: https://github.com/Kong/docker-kong/pull/538
The medium-term thing is to use CentOS 7 for packaging and base images.
And the long-term thing will be to adopt a replacement for CentOS 8 as soon as something suitable comes to light (Rocky Linux maybe).

**Comment by kstephan-wescale** [comment]
And this `sed` hack solves the issue of an EOL base image sufficiently to close here? OK…

**Comment by gszr** [comment]
Hi all, 

With the new builds of docker images coming up (https://github.com/docker-library/official-images/pull/11771), these issues will be solved for Kong 2.5.0, 2.6.1, and 2.7.1 - one will be able to run the regular `yum` commands.

> And this sed hack solves the issue of an EOL base image sufficiently to close here? OK…

As a longer-term solution is still to be designed and implemented, it makes sense to keep this issue open. Thanks, @kstephan-wescale, to bring this up. I will reopen the issue until we have accomplished those milestones.

**Comment by gszr** [comment]
CentOS images are removed by now; officially supported bases include Alpine & Ubuntu
